synapseclient/models/evaluation.py

```diff
@@ -1 +1,970 @@
+            ValueError: If evaluation_id is not set.
+            SynapseHTTPError: If the service rejects the request or an HTTP error occurs.
+
+        Example: Using this function
+        &nbsp;
+
+        ```python
+        from synapseclient.models import Evaluation
+        from synapseclient import Synapse
+        import asyncio
+
+        syn = Synapse()
+        syn.login()
+
+        async def delete_evaluations():
+            # Delete an evaluation by ID
+            await Evaluation(id="9614112").delete_async()
+
+            # Get and then delete an evaluation
+            # First get the evaluation by name, so the ID attribute is set in your
+            # Evaluation object, then delete it.
+            evaluation = await Evaluation(name="My Challenge Evaluation").get_async()
+            await evaluation.delete_async()
+
+        asyncio.run(delete_evaluations())
+        ```
+        """
+        from synapseclient.api.evaluation_services import delete_evaluation
+
+        if not self.id:
+            raise ValueError("id must be set to delete an evaluation")
+
+        await delete_evaluation(
+            evaluation_id=self.id,
+            synapse_client=synapse_client,
+        )
+
+        # Clear the persistent instance since this object has been deleted
+        self._last_persistent_instance = None
+
+    async def get_acl_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> dict:
+        """
+        Get the access control list (ACL) governing this evaluation.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by `Synapse.allow_client_caching(False)` this will use the last created
+                            instance from the Synapse class constructor.
+
+        Returns:
+            The AccessControlList response object as a raw JSON dict.
+
+        Raises:
+            ValueError: If evaluation_id is not set.
+            SynapseHTTPError: If the service rejects the request or an HTTP error occurs.
+
+        Example: Using this function
+        &nbsp;
+
+        ```python
+        from synapseclient.models import Evaluation
+        from synapseclient import Synapse
+        import asyncio
+
+        syn = Synapse()
+        syn.login()
+
+        async def get_evaluation_acl():
+            # Get the evaluation first
+            evaluation = await Evaluation(id="9999999").get_async()
+
+            # Get the ACL for the evaluation
+            acl = await evaluation.get_acl_async()
+            return acl
+
+        acl = asyncio.run(get_evaluation_acl())
+        ```
+        """
+        from synapseclient.api.evaluation_services import get_evaluation_acl
+
+        if not self.id:
+            raise ValueError("id must be set to get evaluation ACL")
+
+        return await get_evaluation_acl(
+            evaluation_id=self.id,
+            synapse_client=synapse_client,
+        )
+
+    async def update_acl_async(
+        self,
```

Comment:
@BryanFauble I wonder if there's an opportunity to create some utils for dealing with ACLs that some/all the model classes can use?